fix(dapps): Detach the pair instructions popup from the pairing popup

alexjba opened the pull request:
### What does the PR do

closes #16887

The pairing popup can be destroyed while the pair instructions popup is active. As a result the pair instructions popup will misbehave. To fix this, the pair instructions component is moved outside of the pair popup scope

### Affected areas

dapps
<!-- List the affected areas (e.g wallet, browser, etc..) -->


https://github.com/user-attachments/assets/110ffc5f-fdaa-4ff3-aad2-b4f1f374b6b9

alexjba commented:
@virginiabalducci One mention here: When a connection proposal is rejected, the same pair url doesn't trigger an error from WC. This is something new. Looks like an SDK bug. Will investigate in another task if other SDK versions behave differently.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 175bd758 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-16893/1/) | 2024-12-05 17:38:48 | ~8 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-16893/1//consoleText) |

virginiabalducci commented:
This looks good!

alexjba commented:
@virginiabalducci I've updated the PR to fix the following flows:

1. Running the flow in the video multiple times per sessions. There was a problem with the instructions popup being opened multiple times in this flow.
2. Added timeouts to avoid the case presented in the bug where reusing a rejected pairing url doesn't have any feedback. I hope there's a proper fix in newer SDK versions, but this would be for the next release maybe. Added a 10 seconds timeout for the url validation and 10 seconds timeout for the pairing. If WC doesn't propose a new connection in 10 seconds after Status initiates a pairing request we'll show the Unknown error message in the pairing modal.

caybro commented on ui/imports/shared/popups/walletconnect/PairWCModal.qml:
```diff
@@ -72,12 +73,7 @@ StatusDialog {
-                Global.openPopup(uriCopyInstructionsPopup)
-            }
-
-            Component {
```
Wouldn't it be cleaner if this was just moved to Popups.qml?

caybro replied:
Fair enough!

caybro reviewed (approved):
LGTM but consider the other option too :)
